VirtoCommerce.CustomerModule.Data/Handlers/LogChangesMemberChangedEventHandler.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using AutoCompare;
+using VirtoCommerce.Domain.Commerce.Model;
+using VirtoCommerce.Domain.Common.Events;
+using VirtoCommerce.Domain.Customer.Events;
+using VirtoCommerce.Domain.Customer.Model;
+using VirtoCommerce.Domain.Customer.Services;
+using VirtoCommerce.Domain.Order.Model;
+using VirtoCommerce.Platform.Core.ChangeLog;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+
+namespace VirtoCommerce.CustomerModule.Data.Handlers
+{
+    public class LogChangesMemberChangedEventHandler : IEventHandler<MemberChangedEvent>
+    {
+        private readonly IChangeLogService _changeLogService;
+        private static readonly string[] ObservedProperties;
+
+        static LogChangesMemberChangedEventHandler()
+        {
+            var memberPropsNames = ReflectionUtility.GetPropertyNames<Contact>(
```

Comment:
I think we can move this configuration to special virtual method, for better extensible if we will want to log extra properties for a contact descendant types.

Reply:
This should happen only once, and with virtual method we would need to put it to the constructor, which is usually not the best idea.
https://www.jetbrains.com/help/resharper/VirtualMemberCallInConstructor.html

Reply:
I didn't say what we need register observed properties in ctor.

Reply:
How else would you do it just once? Now it is done in static ctor. You want to move the configuration to a virtual method. Where would we call it then?

Reply:
We can call this method just before filtering the differences. Another solution is pass the list of observed properties as formal parameters in ctor.

Reply:
The latter approach is better. Let's do that.